Support floats for `fsWatcherDelayS` option

Syncthing supports floats for the setting `fsWatcherDelayS`: https://github.com/syncthing/syncthing/commit/358cf25cffe7b7586f1ded3f0ce27309fec2698e

However, when setting `fsWatcherDelayS` to a float (like `0.5`) in syncthing-android, via the Advanced configuration in the web GUI, it crashes on startup. The log contains (hand-copied):

```
...
Caused by: java.lang.NumberFormatException: Expected an int but was 0.5 at line 137 column 29 path $.defaults.folder.fsWatcherDelayS
...
```

This issue is about supporting floats for `fsWatcherDelayS`. For the crash, I opened another one: https://github.com/Catfriend1/syncthing-android/issues/1366

--- a/app/src/main/java/com/nutomic/syncthingandroid/model/Folder.java
+++ b/app/src/main/java/com/nutomic/syncthingandroid/model/Folder.java
@@ -25,7 +25,7 @@ public class Folder {
     public String path;
     public String type = Constants.FOLDER_TYPE_SEND_RECEIVE;
     public boolean fsWatcherEnabled = true;
-    public int fsWatcherDelayS = 10;
+    public float fsWatcherDelayS = 10;
     private List<SharedWithDevice> devices = new ArrayList<>();
     /**
      * Folder rescan interval defaults to 3600s as it is the default in

--- a/app/src/main/java/com/nutomic/syncthingandroid/util/ConfigXml.java
+++ b/app/src/main/java/com/nutomic/syncthingandroid/util/ConfigXml.java
@@ -392,7 +392,7 @@ public class ConfigXml {
                 // Enable "fsWatcherEnabled" attribute and set default delay.
                 Log.i(TAG, "Set 'fsWatcherEnabled', 'fsWatcherDelayS' on folder " + r.getAttribute("id"));
                 r.setAttribute("fsWatcherEnabled", Boolean.toString(defaultFolder.fsWatcherEnabled));
-                r.setAttribute("fsWatcherDelayS", Integer.toString(defaultFolder.fsWatcherDelayS));
+                r.setAttribute("fsWatcherDelayS", Float.toString(defaultFolder.fsWatcherDelayS));
             }
 
             /**

--- a/app/src/main/java/com/nutomic/syncthingandroid/util/ConfigXml.java
+++ b/app/src/main/java/com/nutomic/syncthingandroid/util/ConfigXml.java
@@ -475,7 +475,7 @@ public class ConfigXml {
             
             folder.type = getAttributeOrDefault(r, "type", Constants.FOLDER_TYPE_SEND_RECEIVE);
             folder.autoNormalize = getAttributeOrDefault(r, "autoNormalize", folder.autoNormalize);
-            folder.fsWatcherDelayS =getAttributeOrDefault(r, "fsWatcherDelayS", folder.fsWatcherDelayS);
+            folder.fsWatcherDelayS = getAttributeOrDefault(r, "fsWatcherDelayS", folder.fsWatcherDelayS);
             folder.fsWatcherEnabled = getAttributeOrDefault(r, "fsWatcherEnabled", folder.fsWatcherEnabled);
             folder.ignorePerms = getAttributeOrDefault(r, "ignorePerms", folder.ignorePerms);
             folder.rescanIntervalS = getAttributeOrDefault(r, "rescanIntervalS", folder.rescanIntervalS);
@@ -592,7 +592,7 @@ public class ConfigXml {
                 r.setAttribute("path", folder.path);
                 r.setAttribute("type", folder.type);
                 r.setAttribute("autoNormalize", Boolean.toString(folder.autoNormalize));
-                r.setAttribute("fsWatcherDelayS", Integer.toString(folder.fsWatcherDelayS));
+                r.setAttribute("fsWatcherDelayS", Float.toString(folder.fsWatcherDelayS));
                 r.setAttribute("fsWatcherEnabled", Boolean.toString(folder.fsWatcherEnabled));
                 r.setAttribute("ignorePerms", Boolean.toString(folder.ignorePerms));
                 r.setAttribute("rescanIntervalS", Integer.toString(folder.rescanIntervalS));

--- a/app/src/main/java/com/nutomic/syncthingandroid/util/ConfigXml.java
+++ b/app/src/main/java/com/nutomic/syncthingandroid/util/ConfigXml.java
@@ -424,6 +424,14 @@ public class ConfigXml {
         }
     }
 
+    private Float getAttributeOrDefault(final Element element, String attribute, Float defaultValue) {
+        try {
+            return element.hasAttribute(attribute) ? Float.parseFloat(element.getAttribute(attribute)) : defaultValue;
+        } catch (NumberFormatException e) {
+            return defaultValue;
+        }
+    }
+
     private String getAttributeOrDefault(final Element element, String attribute, String defaultValue) {
         return element.hasAttribute(attribute) ? element.getAttribute(attribute) : defaultValue;
     }

